### ISSUE TITLE
Standardise server and client code 

### ISSUE THREAD

**Issue by joakee** [QuickRecon/ScotchOARKit#30]
we **_need_** to schedule a cleanup

**Comment by TomW1605** [comment]
would that include making all one language?

**Comment by joakee** [comment]
yes

**Comment by TomW1605** [comment]
if u stay with python or java then i am happy to help after my exams

**Comment by QuickRecon** [comment]
Should we replace the whole thing with a webserver control thing so you just go to a website running on the robot in a browser and controller it from there?

**Comment by TomW1605** [comment]
2 questions
1) what language would u use for the back end and servo stuff?
2) how would u use the controller

other than that the principal sounds good

**Comment by QuickRecon** [comment]
1) node.js with a [serial library](https://github.com/EmergingTechnologyAdvisors/node-serialport)
2) front end javascript [library](http://www.gamepadjs.com/)

**Comment by TomW1605** [comment]
cool btw on the frontend js library [github page](https://github.com/sgraham/gamepad.js/) it says it is deprecated and that it is best just to use the native gamepad api and the gamepad api looks a lot easier that jinput

**Comment by QuickRecon** [comment]
ok

**Comment by joakee** [comment]
yeah, lets do a web thing, i always had this in mind since we started

**Comment by Whoa-exclamation-mark** [comment]
I was talking to the teacher before he went on LSV and I was discussing to get rid of the Pi and change it to a ESP8266 instead because the PI is way to powerful for what we need (to expensive) and annoying to set up. So we would be having a java front end (client) and a C++ backend (server).

**Comment by joakee** [comment]
No. We simply cannot do that. It's already a problem that we have deviated from the OARKit initiative so much, and moving to a completely new hardware platform will not only add more inconveniences on top of what we already have, but it will also hinder our ability to work with the wider OARKit community. I am not sure if Edwards realises this, but robotics is actually just Scotch's branch of the OARKit. I spoke to Raymond some time ago about replacing the Pi with an Arduino, he said that would be a foolish thing to do. I agree with him. Using an Arduino limits our ability to develop the OARKit as part of the initiative (which we definetly should be doing), and using a Raspberry Pi greatly expands our horizons in terms of scaling, new features and system expandability. The only reason the Pi is so hard to set up is because of how complex the new Java code has been. If we just kept the Python, (a working and easily expandable codebase) and built upon it to incorporate these new features; not only will it run on the Pi without annoyance, it would be less needlessly complex as the Java code isn't. Trust me. I am the only one who has been here since the beginning and I know what this is really about.

**Comment by TomW1605** [comment]
i agree with oakey and another consideration is that a Pi can run almost any language so you are not limited to unfamiliar or weak languages.

also campbell are u sure u got the name of the pi alternative right? when i [googled](https://www.google.com.au/search?q=ESP88226&oq=ESP88226&aqs=chrome..69i57j69i59.1449j0j7&sourceid=chrome&ie=UTF-8) it there was not much there.

**Comment by Whoa-exclamation-mark** [comment]
I fixed the name of the module (I had a 8 in their by accident and put the newer one that has not been fully made... it is the ESP8266). This will help us by allowing us to do micro-controller engineering and also we can directly use the Dynamixel library instead of having the Raspberry Pi in the middle of the communications. Anyway it is nice to learn C++ :)

**Comment by Whoa-exclamation-mark** [comment]
Code clean up is issue #28. This issue is only about standardise code on the server and client
